File: bst.cs
using System;

class Node {
    public int Value;
    public Node Left, Right;

    public Node(int value) {
        Value = value;
    }
}

class BST {
    private Node root;

    public void Insert(int value) {
        root = InsertRec(root, value);
    }

    private Node InsertRec(Node node, int value) {
        if (node == null) return new Node(value);

        if (value < node.Value)
            node.Left = InsertRec(node.Left, value);
        else if (value > node.Value)
            node.Right = InsertRec(node.Right, value);

        return node;
    }

    public void Inorder() {
        InorderRec(root);
        Console.WriteLine();
    }

    private void InorderRec(Node node) {
        if (node == null) return;
        InorderRec(node.Left);
        Console.Write(node.Value + " ");
        InorderRec(node.Right);
    }

    static void Main() {
        BST tree = new BST();
        int[] values = {50,30,70,20,40,60,80};

        foreach (int v in values)
            tree.Insert(v);

        tree.Inorder();
    }
}

File: bst.go
package main

import "fmt"

type Node struct {
	value int
	left  *Node
	right *Node
}

type BST struct {
	root *Node
}

func (t *BST) Insert(value int) {
	t.root = insertRec(t.root, value)
}

func insertRec(node *Node, value int) *Node {
	if node == nil {
		return &Node{value: value}
	}
	if value < node.value {
		node.left = insertRec(node.left, value)
	} else if value > node.value {
		node.right = insertRec(node.right, value)
	}
	return node
}

func inorder(node *Node) {
	if node == nil {
		return
	}
	inorder(node.left)
	fmt.Print(node.value, " ")
	inorder(node.right)
}

func main() {
	tree := BST{}
	values := []int{50, 30, 70, 20, 40, 60, 80}

	for _, v := range values {
		tree.Insert(v)
	}

	fmt.Println("Inorder:")
	inorder(tree.root)
}
